app/controllers/concerns/saml_idp_logout_concern.rb

```diff
@@ -56,6 +56,9 @@ def resource_in_slo?(resource)
 
   def generate_slo_request(resource)
     slo_identity = fetch_identity_for_slo(resource)
+    # The SP has not implemented SLO; do not generate a request!
+    return nil if
+      slo_identity.sp_metadata[:assertion_consumer_logout_service_url].nil?
     {
       message: slo_request_builder(
         slo_identity.sp_metadata,
```

Comment:
Since this results in a Code Climate issue, what do you think of this alternate solution:

- Add the logic to `saml_logout_message` by defining a new `slo_not_implemented_at_sp?` method, like this:

```ruby
def saml_logout_message
  return handle_saml_logout_response if successful_saml_response?
  return nil if failed_saml_response? || slo_not_implemented_at_sp?
  return handle_saml_logout_request(name_id_user) if valid_saml_request?

  generate_slo_request(current_user) if user_signed_in_and_has_identity?
end

private

def slo_not_implemented_at_sp?
  return false unless current_user.present?

  slo_identity = fetch_identity_for_slo(current_user)
  slo_identity.sp_metadata[:assertion_consumer_logout_service_url].nil?
end
```

By returning nil early, it makes it clearer in which scenarios the logout should finish at the IdP right away, based on the line `return finish_slo_at_idp if message.nil?` in the `logout` action, as opposed to adding the logic in the `generate_slo_request` method, whose name implies that an SLO request will actually be generated.